perf-tool/src/monitor.cpp

```diff
@@ -170,6 +170,25 @@ JNIEXPORT void JNICALL MonitorContendedEntered(jvmtiEnv *jvmtiEnv, JNIEnv *env,
         check_jvmti_error(jvmtiEnv, error, "Unable to deallocate thread name.\n");
         // Local JNI refs to threadInfo.thread_group and threadInfo.context_class_loader will be freed upon return
     }
+    /* Get Java thread ID */
+    jclass threadClass = env->FindClass("java/lang/Thread");
+    if (threadClass)
+    {
+        jmethodID getIdMethod = env->GetMethodID(threadClass, "getId", "()J");
+        if (getIdMethod)
+        {
+            jlong tid = env->CallLongMethod(thread, getIdMethod);
```

Comment:
Don't you need to call this with a thread object?

Reply:
Ohh missed that, I saw the call the `FindClass` and for some reason thought we were passing in `threadClass`.